embed/config.go

```diff
@@ -50,8 +47,25 @@ var (
 	ErrConflictBootstrapFlags = fmt.Errorf("multiple discovery or bootstrap flags are set. " +
 		"Choose one of \"initial-cluster\", \"discovery\" or \"discovery-srv\"")
 	ErrUnsetAdvertiseClientURLsFlag = fmt.Errorf("--advertise-client-urls is required when --listen-client-urls is set explicitly")
+
+	DefaultListenPeerURLs           = "http://localhost:2380"
+	DefaultListenClientURLs         = "http://localhost:2379"
+	DefaultInitialAdvertisePeerURLs = "http://localhost:2380"
+	DefaultAdvertiseClientURLs      = "http://localhost:2379"
 )
 
+func init() {
+	ip, err := netutil.GetDefaultHost()
+	if err != nil {
+		return
+	}
+	// found default host; listen everywhere, advertise on default host
+	DefaultListenPeerURLs = "http://0.0.0.0:2380"
```

Comment:
i do not think we should listen on wide by default even if we found the default host for security reasons. by finding default adv url, users do not need to set avd ulrs themselves but not listen url I think.